x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/DataFrameFeatureSet.java

```diff
@@ -77,6 +78,8 @@ public void usage(ActionListener<XPackFeatureSet.Usage> listener) {
         }
 
         GetDataFrameTransformsAction.Request transformsRequest = new GetDataFrameTransformsAction.Request(MetaData.ALL);
+        // TODO Before 7.1, see https://github.com/elastic/elasticsearch/issues/39994
+        transformsRequest.setPageParams(new PageParams(0, 10_000));
```

Comment:
we should start with a lower limit, we can increase it after verification. At the moment there is no reason to support 10k as this is an unreleased feature.

`setPageParams` is unchecked, we should add a check there, if not upstream we can override it and enforce a lower limit in `GetDataFrameTransformsAction`.

Reply:
So, you want to lock down the limit below what the ES limit is for the `PageParams` as a whole? Did you have anything in mind?

Reply:
maybe `1000` or `500`? My worst case guesstimate: If a config object requires max 10 kb, a request for 1000 would require roughly max 10 MB. - still no idea if anyone ever wants 1000 in 1 request.

Maybe off-topic: Do we have a default? E.g. search returns default 10 hits, do we have something similar? In this case we could also return 10 as default but allow 1000 max. Good compromise?

Reply:
@hendrikmuhs our default is currently 100, but can easily change to 10. max of 1000 seems reasonable.

Reply:
sounds good, I have no strong feelings whether it is 10 or 100, only 1000 sounds to much to me.

Reply:
Refactored so that this type of query is not even necessary any longer. We don't really care about individual configs, just the total number of them. This way we can do `total_configs - total_tasks = num_configs_without_tasks` and just add that many `STOPPED` state counts.